### PR TITLE
Update message manager and signature controller

### PR DESCRIPTION
**Pull request by Gudahtt**
## Explanation

The packages `@metamask/message-manager` and
`@metamask/signature-controller` have been updated to the latest versions. These versions were part of part of the [core monorepo v53](MetaMask/core#1385) release. The remaining packages released as part of v53 will be updated in later PRs.

These releases included a few breaking changes, but none that affect the extension. Both packages now expect `getCurrentChainId` to return `Hex`, but that was already the case here. Additionally the signature controller now has a peer dependency on the approval controller v3, which is already present and at the correct version.

One change listed as non-breaking did impact the extension anyway, which is the addition of the optional `metadata` property to the `AbstractMessage` type. This resulted in a type error in  the `decrypt-message` and `encryption-public-key` controllers, because this type was used for the controller state and optional properties are not allowed. They have been updated to omit this property for now, since it's not used. This resolves the type error.

Relates to #19271

## Manual Testing Steps

No functional changes

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by Gudahtt** [comment]
Found a type error. Moved to draft while I investigate.

**Review by mcmire — DISMISSED** [review]
Looks good!

File: app/scripts/controllers/encryption-public-key.ts
Comment (Karlitos88) [inline app/scripts/controllers/encryption-public-key.ts:1]: app/scripts/controllers/encryption-public-key.ts public Keys congratulacions thanks

**Review by Karlitos88 — APPROVED** [review]
Thanks program inteligent revolution identifier factory checking changes verify and cheout changes normalize extensions autoritze

**Comment by Gudahtt** [comment]
... I don't know what happened with this PR, it's showing tons of commits for some reason. And now it's not responding to force pushes. Maybe another GitHub service failure.

**Comment by mcmire** [comment]
@Gudahtt Maybe recreate this PR? 😬

**Comment by Gudahtt** [comment]
I asked GitHub Support and they suggested changing the base branch as a way to re-sync the PR. Seemed to work!

**Review by mcmire — DISMISSED** [review]

**Review by mcmire — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Pull request merged**